### PR TITLE
Enable private field promotion for flutter_tools

### PR DESCRIPTION
**Pull request by goderbauer**
New feature in upcoming Dart 3.2. See https://github.com/dart-lang/language/issues/2020. Feature is enabled by bumping the min SDK version to 3.2.

Part of https://github.com/flutter/flutter/issues/134476.

### CODE REVIEW

**Comment by christopherfujino** [comment]
Woohoo, excited about this!

File: packages/flutter_tools/test/commands.shard/hermetic/generate_localizations_test.dart [hunk 1]
```diff
@@ -466,7 +466,7 @@ format: true
     pubspecFile.writeAsStringSync('''
   name: test
   environment:
-    sdk: '>=3.0.0-0 <4.0.0'
+    sdk: '>=3.2.0-0 <4.0.0'
```
Comment (christopherfujino) [inline packages/flutter_tools/test/commands.shard/hermetic/generate_localizations_test.dart:469]: did you just grep to find all these magic strings? outside of this PR, maybe we should just have a global const string to make bumping this easier.

**Review by christopherfujino — APPROVED** [review]
LGTM. Thanks for this one!

File: packages/flutter_tools/test/commands.shard/hermetic/generate_localizations_test.dart (hunk 1 above)
Comment (goderbauer) [reply packages/flutter_tools/test/commands.shard/hermetic/generate_localizations_test.dart:469]: That's what I did, yeah. That sounds like a good idea!

**Pull request merged**